### PR TITLE
Auth config

### PR DESCRIPTION
**Pull request by Gil-AdB**

### CODE REVIEW

File: login.go [hunk 1]
```diff
@@ -39,69 +42,116 @@ func (cmd *loginCommand) Register(fs *flag.FlagSet) {
...
+
+	if err := json.NewDecoder(bytes.NewReader(buf)).Decode(&result); err != nil {
+		return result, err
+	}
+
+	return result, nil
+}
+
+
+func strinpEndlines(contents string) string {
```
Comment (dorkolog) [inline login.go:74]: ```stripEndlines``` ?

File: login.go [hunk 2]
```diff
@@ -39,69 +42,116 @@ func (cmd *loginCommand) Register(fs *flag.FlagSet) {
...
-		authConfig.IdentityToken = token
+	if authConfig.IdentityToken != "" {
+		token, err := r.Token(r.URL)
+		if err != nil && err != registryapi.ErrBasicAuth {
+			return fmt.Errorf("getting registry token failed: %v", err)
+		}
+
+		// Configure the token.
+		if token != "" {
+			authConfig.Password = ""
```
Comment (dorkolog) [inline login.go:152]: i think you should also set ```authConfig.Username = ""```

File: login.go [hunk 3]
```diff
@@ -39,69 +42,116 @@ func (cmd *loginCommand) Register(fs *flag.FlagSet) {
...
-	token, err := r.Token(r.URL)
-	if err != nil && err != registryapi.ErrBasicAuth {
-		return fmt.Errorf("getting registry token failed: %v", err)
-	}
 
-	// Configure the token.
-	if token != "" {
-		authConfig.Password = ""
-		authConfig.IdentityToken = token
+	if authConfig.IdentityToken != "" {
```
Comment (dorkolog) [inline login.go:144]: please revert the if statement and return error if its empty

File: login.go [hunk 4]
```diff
@@ -39,69 +42,116 @@ func (cmd *loginCommand) Register(fs *flag.FlagSet) {
...
+		}
+
+		// Configure the token.
+		if token != "" {
+			authConfig.Password = ""
+			authConfig.IdentityToken = token
+		}
 	}
 
 	// Save the config value.
```
Comment (dorkolog) [inline login.go:157]: I think this can be removed as we are using it inside a container and don't wish to create side-effect

File: login.go [hunk 5]
```diff
@@ -39,69 +42,116 @@ func (cmd *loginCommand) Register(fs *flag.FlagSet) {
...
+			cmd.serverAddress = args[0]
 		}
 
+		// Set the default registry server address.
+		if cmd.serverAddress == "" {
+			cmd.serverAddress = defaultDockerRegistry
+		}
+
+		// Get the auth config.
+		dcfg, authConfig, err = configureAuth(cmd.user, cmd.password, cmd.serverAddress)
```
Comment (dorkolog) [inline login.go:119]: please see my comment about removing ```dcfg``` support

File: login.go (hunk 2 above)
Comment (Gil-AdB) [reply login.go:152]: This was taken as-is from the original code, apart from indentation, So I don't want to make a change without a good reason.

File: login.go (hunk 3 above)
Comment (Gil-AdB) [reply login.go:144]: Irrelevant

File: login.go (hunk 4 above)
Comment (Gil-AdB) [reply login.go:157]: This is the login process. The pull/push process needs the saved credentials in order to authenticate to the remote registry.

Not saving the credentials will result in an error when trying to pull images.

**Pull request merged**